app/store/price-context.tsx

```diff
@@ -0,0 +1,97 @@
+import { saveJson, loadJson } from "@app/utils/storage"
+import { createContext, useContext, useEffect, useReducer } from "react"
+import * as React from "react"
+import { usePriceSubscription } from "@app/hooks/use-price-subscription"
+
+export type PriceData =
+  | {
+      initialized: false
+    }
+  | {
+      price: number
+      priceDate: Date
+      initialized: true
+    }
+
+type PriceContextType = {
+  priceData: PriceData
+  setPrice: (price: number, priceDate?: Date) => void
+}
+
+type PriceReducerAction = {
+  type: "set_price"
```

Comment:
Not a big deal but this string should be a const which can be exported so that when we are firing actions from other components we avoid typos which could cause the event not to fire.